compiler/resolution/functionResolution.cpp

```diff
@@ -71,18 +72,20 @@ class DisambiguationState {
 public:
         DisambiguationState();
 
-  void  updateParamPrefers(int                          preference,
-                           const char*                  argStr,
-                           const DisambiguationContext& DC);
-
   bool  fn1MoreSpecific;
   bool  fn2MoreSpecific;
 
   bool  fn1Promotes;
   bool  fn2Promotes;
 
-  // 1 == fn1, 2 == fn2, -1 == conflicting signals
-  int   paramPrefers;
+  bool fn1WeakPreferred;
+  bool fn2WeakPreferred;
+
+  bool fn1WeakerPreferred;
+  bool fn2WeakerPreferred;
+
+  bool fn1WeakestPreferred;
+  bool fn2WeakestPreferred;
```

Comment:
I was expecting to explain that in the specification. But at a high level, it's because something like 'param x:int(16) = 5' should be able to coerce to 'int' or 'uint' or 'int(8)' or .... Meanwhile 'param y = 5' should be able to coerce to 'int(16)' or 'uint' or 'int(8)' or ....  Now imagine we are resolving 'x+y'. We want it to resolve to the 'int(16)' version because 'x' has a type specified, but 'y' is a default type. It used to be the case that this version was chosen simply because non-default-sized ints didn't allow param conversion.